test-utils/state-viewer/src/apply_chain_range.rs

```diff
@@ -0,0 +1,288 @@
+                inc_and_report_progress(&processed_blocks_cnt);
+                return;
+            };
+
+            let mut chain_store_update = ChainStoreUpdate::new(&mut chain_store);
+            let receipt_proof_response = chain_store_update
+                .get_incoming_receipts_for_shard(
+                    shard_id,
+                    block_hash,
+                    prev_block.chunks()[shard_id as usize].height_included(),
+                )
+                .unwrap();
+            let receipts = collect_receipts_from_response(&receipt_proof_response);
+
+            let chunk_inner = chunk.cloned_header().take_inner();
+            let is_first_block_with_chunk_of_version =
+                check_if_block_is_first_with_chunk_of_version(
+                    &mut chain_store,
+                    runtime_adapter.as_ref(),
+                    block.header().prev_hash(),
+                    shard_id,
+                )
+                    .unwrap();
+            runtime_adapter
+                .apply_transactions(
+                    shard_id,
+                    chunk_inner.prev_state_root(),
+                    None,
+                    height,
+                    block.header().raw_timestamp(),
+                    block.header().prev_hash(),
+                    block.hash(),
+                    &receipts,
+                    chunk.transactions(),
+                    chunk_inner.validator_proposals(),
+                    prev_block.header().gas_price(),
+                    chunk_inner.gas_limit(),
+                    &block.header().challenges_result(),
+                    *block.header().random_value(),
+                    true,
+                    is_first_block_with_chunk_of_version,
+                    None,
+                )
+                .unwrap()
+        } else {
+            let chunk_extra = chain_store.get_chunk_extra(block.header().prev_hash(), &shard_uid).unwrap().clone();
+            prev_chunk_extra = Some(chunk_extra.clone());
+
+            runtime_adapter
+                .apply_transactions(
+                    shard_id,
+                    chunk_extra.state_root(),
+                    None,
+                    block.header().height(),
+                    block.header().raw_timestamp(),
+                    block.header().prev_hash(),
+                    &block.hash(),
+                    &[],
+                    &[],
+                    chunk_extra.validator_proposals(),
+                    block.header().gas_price(),
+                    chunk_extra.gas_limit(),
+                    &block.header().challenges_result(),
+                    *block.header().random_value(),
+                    false,
+                    false,
+                    None,
+                )
+                .unwrap()
+        };
+
+        println!("outcomes: {:#?}", apply_result.outcomes);
```

Comment:
It feels a bit odd that we use all three of `tracing`, `indicatiff`, `println` to print stuff to the terminal. I'd probaby try to stick to prints for all output (but, again, not that it matters a lot)